lib/internal/quic/core.js

```diff
@@ -2673,7 +2673,14 @@ class QuicStream extends Duplex {
 
     // Put the QuicStream into detached mode before calling destroy
     this[kSetHandle]();
-    this.destroy();
```

Comment:
I think just removing this line would be the samething if `autoDestroy` is enabled.

Reply:
Also I think the current code is probably correct, kDestroy is not supposed to be a graceful shutdown? @jasnell?

Reply:
`autoDestroy` is indeed enabled
`kDestroy()` is called from `onSessionClose()` which is called from C++ `QuicSession::Close()`, it handles both errors and a normal close

Reply:
I'm pretty sure that the code below is unnecessary and removing `destroy()` here might or might not be correct.

Reply:
node test/parallel/test-quic-simple-server-uni.js

You need a QUIC-enabled build

Reply:
The test was working before you stopped sending the data from destroyed streams.
In QUIC you can mark a data chunk as being the final data chunk. In this case the client is supposed to close the connection upon receiving it. The QUIC code will immediately `destroy` the `Readable` in a `nextTick` handler. When using `on('data')` the user code will get a chance to read the last chunk, but when using the async iterator, and after your change, the Promise won't be resolved until the `'close'` is processed.

Reply:
@jasnell said that a complete shake-up on that code was in the works, so he probable doesn't care that much :smile: 
But I had already debugged it before he said it

Reply:
Indeed. Starting likely week after next this entire part of the code is going to be refactored entirely away from the Duplex interface. I don't have time to detail the specifics at the moment but will do so soon.